src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi

```diff
@@ -384,16 +397,21 @@ cdef class _AioCall(GrpcCallWrapper):
             send_close_op,
         )
 
-        # Sends out the request message.
-        await execute_batch(self,
-                            outbound_ops,
-                            self._loop)
-
-        # Receives initial metadata.
-        self._set_initial_metadata(
-            await _receive_initial_metadata(self,
-                                            self._loop),
-        )
+        try:
+            # Sends out the request message.
+            await execute_batch(self,
+                                outbound_ops,
+                                self._loop)
+
+            # Receives initial metadata.
+            self._set_initial_metadata(
+                await _receive_initial_metadata(self,
+                                                self._loop),
+            )
+        except ExecuteBatchError as batch_error:
+            # Core should explain why this batch failed
+            await status_task
+            assert self._status.code() != StatusCode.ok
```

Comment:
what does this provide besides of doing an `assert`?

Reply:
It guards our implementation that, if any batch failed in Core, our layer should not return an `OK` status. Otherwise, there is a bug somewhere.

Reply:
Having the feeling that the main purpose of this exception is not progressing the exception to the up layers, I guess that is what the following sentence tries to explain

> # Core should explain why this batch failed

Regarding guards, how much value is adding to us? does it mean that guards should be/can be explicitly used during the code? if not, why in that use case is so important?

Reply:
Assertion lifted. This guard should not be used, it is more or less a placebo. After the batch failure, the code should enter error path.